### ISSUE TITLE
Horizontal scrolling in quickpick

### ISSUE THREAD

**Issue by karrtikr** [microsoft/vscode#101874]
Hi,
Is there a way to enable horizontally scrolling inside the quickpick? (I have looked at quickpick options but wasn't able to find any) Sometimes the text may exceed the horizontal length.

![image](https://user-images.githubusercontent.com/13199757/86831000-35728800-c04b-11ea-8c65-be14775d0ce5.png)

**Comment by chrmarti** [comment]
We have avoided horizontal scrollbars as much as possible in the UI (the editor area being the big exception). I can investigate if we can show a tooltip when hovering with the mouse.

**Comment by karrtikr** [comment]
> We have avoided horizontal scrollbars as much as possible in the UI

I see ok. Btw why is that the case?

 Same goes for command palette as well,

![image](https://user-images.githubusercontent.com/13199757/86895495-67bace80-c0b9-11ea-82b6-c538faaf19ba.png)

**Comment by chrmarti** [comment]
IMO the need for horizontal scrollbars indicates that the content should be rearranged to better fit a vertical flow. Having only one direction to scroll makes it easier to use. Once you do scroll horizontally the visible content often looks unaligned because its horizontal begin has scrolled out (that is much less of a problem with vertical scrolling).

The command palette uses the same underlying widget.

**Comment by daniv-msft** [comment]
This issue is problematic for accessibility reasons: a lot of extensions use quick picks and, as soon as the user starts zooming a bit, part of the text isn't readable anymore.
I liked the suggestion of adding a tooltip. Is this something that might be possible to add?

**Comment by fsteffi** [comment]
GitHubTags:#A11yMAS;#A11yTCS;#Win32;#DesktopApp;#A11y_VSCode;#WCAG1.3.4;#A11ySev2;#VisualStudioCode-Win32-Apr2021;#Benchmark;#Visual Studio Code Client;#Orientation;#a11y-partner;#A11yWCAG2.1;

**Comment by TylerLeonhardt** [comment]
Yep I like that a lot.

**Comment by isidorn** [comment]
After discussion with @Tyriar and @TylerLeonhardt we realised that the `QuickInput` can not have a dependency on the `HoverService` because it lives in `base` and there we do not have services at all. Because of this technical limitation we agreed that we should start using a native hover for this because:
* Native hover's size depends on the screen display scale, so users can change this to increas the native hover size
* Native hover works well with magnifiers

Once we do this and if we hear feedback from users that a Custom hover is more fitting because it works with VS Code zoom than as a workaround we can pass in a hover delegate callback to `QuickInput`  which is injected via workbench. Though this is nothing for now, as it would introduce a hack without any user feedback which is not what we want.

**Comment by TylerLeonhardt** [comment]
verification steps:
```ts
        await window.showQuickPick([{
		label: 'label',
		detail: 'asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf asdf'
	}], { matchOnDetail: true });
```

Ensure:
* when you hover over the detail, a tooltip is shown
* matching on the detail still works (type `asdf` in the input, highlights should be shown in the detail)
* now add an icon like `$(loading-spin)` or `$(star)` to the detail string

**Comment by rchiodo** [comment]
/verified?

Is the tooltip supposed to show the icon? I didn't think so because it mentions a native tooltip, but wasn't sure.

**Comment by TylerLeonhardt** [comment]
@rchiodo that's covered in https://github.com/microsoft/vscode-internalbacklog/issues/2488

Thanks for catching that!
